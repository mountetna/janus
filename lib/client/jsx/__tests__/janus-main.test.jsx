import React from 'react';
import {Provider} from 'react-redux';
import { act, create } from 'react-test-renderer';
import {mockStore} from '../test-helpers';
import {stubUrl} from 'etna-js/spec/helpers';
import JanusMain from '../janus-main';

describe('JanusMain', () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      user: {
	email: "janus@two-faces.org",
	name: "Janus Bifrons",
	permissions: { }
      }
    });
  });

  it('renders', async () => {
    const initialStubs = [
      stubUrl({
        verb: 'get',
	path: '/api/user/projects',
        status: 200,
        response: {projects: [
          {
            project_name: "tunnel",
            project_name_full: "Tunnel",
            role: "viewer",
            privileged: true
          }, {
            project_name: "mirror",
            project_name_full: "Mirror",
            role: "editor",
            privileged: null
          }, {
            project_name: "gateway",
            project_name_full: "Gateway",
            role: "editor",
            privileged: null
          } ]}
      })
    ];

    let component = create(
      <Provider store={store}>
        <JanusMain/>
      </Provider>
    );

    await act( async () => {
        await new Promise((resolve) => setTimeout(resolve, 15));
    });

    expect( component.toJSON() ).toMatchSnapshot()
  });
});
