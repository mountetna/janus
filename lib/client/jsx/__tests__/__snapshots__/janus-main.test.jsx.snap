// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JanusMain renders 1`] = `
<div
  id="janus-main"
>
  <div
    id="identity-group"
  >
    <div
      className="title"
    >
      Your Identity
    </div>
    <div
      className="item"
    >
      Janus Bifrons
    </div>
    <div
      className="item"
    >
      janus@two-faces.org
    </div>
    <div
      className="item"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Copy Token
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  <div
    id="user-projects"
  >
    <div
      className="title"
    >
      Your Projects
    </div>
    <div
      className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <table
        aria-label="project users"
        className="MuiTable-root"
        role={null}
      >
        <thead
          className="MuiTableHead-root"
          role={null}
        >
          <tr
            className="MuiTableRow-root MuiTableRow-head"
            role={null}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              Project Name
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              Title
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              Role
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-paddingCheckbox"
              scope="col"
            >
              Privileged
            </th>
          </tr>
        </thead>
        <tbody
          className="MuiTableBody-root"
          role={null}
        />
      </table>
    </div>
  </div>
</div>
`;
